### PR TITLE
Static build

### PR DESCRIPTION
**Pull request by davidd-amd**
Updates to CMake support to enable static builds and static-dev packaging. Moves from `namespace Tensile` to `namespace TensileLite` to prevent name collisions in downstream consumers that depend on Tensile and TensileLite.

Ran `hipblaslt-test` and `rocblas-test --gtest_filter="*hipblaslt*"`.

### CODE REVIEW

**Review by TorreZuk — DISMISSED** [review]
I would suggest you remove the actual static build changes from this PR.  First get approval and merge only the namespace changes (which may be easy to get approval for).   Assuming none of the namespace is exported in public API I approve.    Otherwise the comments I would make relate to only static build runtime issues...

**Comment by davidd-amd** [comment]
Should we add any docs changes to this PR or separate?

**Comment by TorreZuk** [comment]
> Should we add any docs changes to this PR or separate?

Yes docs should go in with features.   I would use noCI until you are close to merge if you have docker to test now.

**Comment by davidd-amd** [comment]
> > Should we add any docs changes to this PR or separate?
> 
> Yes docs should go in with features. I would use noCI until you are close to merge if you have docker to test now.

Here is the docs requirement: Document how to deploy and use runtime code objects from test application statically linked to hibplasLt. 

tensile_host.cpp has logic for finding the co files and users can set `HIPBLALT_TENSILE_LIBPATH` if tensile_host cannot find them.

**Comment by TorreZuk** [comment]
Look at current rocBLAS docs on this https://github.com/ROCm/rocBLAS-internal/blob/develop/docs/install/Linux_Install_Guide.rst#static-library, which would also need details added in rocBLAS PR or maybe link to hipblasLt section

File: docs/installation.rst
Comment (amd-jnovotny) [inline docs/installation.rst]:
```suggestion
To build static libraries with ``install.sh``, use ``--static``.
This produces a non-standard static library build. This means it has an additional runtime dependency 
consisting of the entire ``hipblaslt/`` subdirectory, which is located in the ``/opt/rocm/lib`` folder. 
You can relocate this folder, but you must set the environment variable ``HIPBLASLT_TENSILE_LIBPATH``
accordingly.
```

**Review by amd-jnovotny — DISMISSED** [review]
Just a few editorial suggestions for the docs component.

**Comment by davidd-amd** [comment]
@jichangjichang @KKyang this PR is ready for review. The following tests are failing in the static pipelines but it looks like they are failing for the other builds as well:

Tensile/Tests/common/groupedgemm/grouped_gemm.yaml
Tensile/Tests/common/sparse/i8_activation.yaml

**Comment by davidd-amd** [comment]
@jichangjichang @KKyang would either of you mind providing review or recommending someone for review?

**Comment by davidd-amd** [comment]
> I would suggest you remove the actual static build changes from this PR. First get approval and merge only the namespace changes (which may be easy to get approval for). Assuming none of the namespace is exported in public API I approve. Otherwise the comments I would make relate to only static build runtime issues...

FYI The namespace changes were separated into another PR and are already merged.

**Review by KKyang — APPROVED** [review]

**Review by bstefanuk — APPROVED** [review]

**Pull request merged**
